File: game_server/test/sc/server/plugins/TestGameState.java
package sc.server.plugins;

import sc.shared.PlayerColor;

public class TestGameState {
  public Integer round = 0;
  public int state = 0;
  public int lastPlayerIndex = 0;
  public int turn;
  public PlayerColor currentPlayer;

  public PlayerColor startPlayer;


  public TestPlayer red;
  public TestPlayer blue;


  public TestGameState() {
    this.turn = 0;
    this.currentPlayer = PlayerColor.RED;
    this.startPlayer = PlayerColor.RED;
    this.red = new TestPlayer(PlayerColor.RED);
    this.blue = new TestPlayer(PlayerColor.BLUE);
  }

  /** wechselt den Spieler, der aktuell an der Reihe ist anhand von {@code turn} */
  public void switchCurrentPlayer() {
    if (turn % 2 == 0) {
      currentPlayer = PlayerColor.RED;
    } else {
      currentPlayer = PlayerColor.BLUE;
    }
  }

}

File: server/test/sc/server/plugins/TestGameState.kt
package sc.server.plugins

import sc.api.plugins.IGameState
import sc.framework.plugins.Player
import sc.server.helpers.TestTeam

data class TestGameState(
        override var turn: Int = 0,
        var state: Int = 0,
): IGameState {
    override val currentTeam: TestTeam
        get() = TestTeam.values()[turn % TestTeam.values().size]
    override val round get() = turn / 2
    val red = Player(TestTeam.RED)
    val blue = Player(TestTeam.BLUE)
    
    override fun clone() = TestGameState(turn, state)
}
